fetch-publication/src/test/java/no/unit/nva/publication/fetch/FetchPublicationHandlerTest.java

```diff
@@ -75,52 +78,58 @@ public void setUp() {
+        assertTrue(gatewayResponse.getHeaders().containsKey(ACCESS_CONTROL_ALLOW_ORIGIN));
     }
 
     @Test
     @DisplayName("handler Returns NotFound Response On Publication Missing")
     public void handlerReturnsNotFoundResponseOnPublicationMissing() throws IOException, ApiGatewayException {
-        when(publicationService.getPublication(any(UUID.class)))
-            .thenThrow(new NotFoundException("Error"));
+        when(publicationService.getPublication(any(UUID.class))).thenThrow(new NotFoundException("Error"));
 
-        fetchPublicationHandler.handleRequest(inputStream(), output, context);
+        fetchPublicationHandler.handleRequest(generateHandlerRequest(), output, context);
 
-        GatewayResponse gatewayResponse = objectMapper.readValue(output.toString(), GatewayResponse.class);
+        GatewayResponse<PublicationResponse> gatewayResponse = objectMapper.readValue(output.toString(),
+                PARAMETERIZED_GATEWAY_RESPONSE_TYPE);
         assertEquals(SC_NOT_FOUND, gatewayResponse.getStatusCode());
-        assertTrue(gatewayResponse.getHeaders().keySet().contains(CONTENT_TYPE));
-        assertTrue(gatewayResponse.getHeaders().keySet().contains(ACCESS_CONTROL_ALLOW_ORIGIN));
+        assertTrue(gatewayResponse.getHeaders().containsKey(CONTENT_TYPE));
+        assertTrue(gatewayResponse.getHeaders().containsKey(ACCESS_CONTROL_ALLOW_ORIGIN));
     }
 
     @Test
     @DisplayName("handler Returns BadRequest Response On Empty Input")
     public void handlerReturnsBadRequestResponseOnEmptyInput() throws IOException {
-        InputStream input = new HandlerUtils(objectMapper)
-            .requestObjectToApiGatewayRequestInputSteam(null, null);
-        fetchPublicationHandler.handleRequest(input, output, context);
+        InputStream inputStream = new HandlerRequestBuilder<InputStream>(objectMapper)
+                .withBody(null)
+                .withHeaders(null)
+                .withPathParameters(null)
+                .build();
+        fetchPublicationHandler.handleRequest(inputStream, output, context);
 
-        GatewayResponse gatewayResponse = objectMapper.readValue(output.toString(), GatewayResponse.class);
+        GatewayResponse<PublicationResponse> gatewayResponse = objectMapper.readValue(output.toString(),
+                PARAMETERIZED_GATEWAY_RESPONSE_TYPE);
         assertEquals(SC_BAD_REQUEST, gatewayResponse.getStatusCode());
+        assertThat(gatewayResponse.getBody(), containsString(IDENTIFIER_NULL_ERROR));
     }
 
     @Test
     @DisplayName("handler Returns BadRequest Response On Missing Path Param")
     public void handlerReturnsBadRequestResponseOnMissingPathParam() throws IOException {
-        Map<String, Object> event = Map.of(HEADERS, Map.of());
-        InputStream inputStream = new ByteArrayInputStream(objectMapper.writeValueAsBytes(event));
-
+        InputStream inputStream = new HandlerRequestBuilder<InputStream>(objectMapper)
```

Comment:
This is a one level of abstraction lower than the rest of the code. refactor to a method